src/libYARP_sig/include/yarp/sig/Vector.h

```diff
@@ -277,7 +308,28 @@ class yarp::sig::VectorOf : public VectorBase
         bytes.allocateOnNeed(bytes.used()+sizeof(T),bytes.length()*2+sizeof(T));
         bytes.setUsed(bytes.used()+sizeof(T));
         _updatePointers();
-        first[len-1] = elem;
+        new (&((*this)[len-1])) T(elem); // non-allocating placement operator new
```

Comment:
As discussed, we should cherry-pick this change in master

Reply:
Also this c1802a3